ultra/baselines/ddpg/ddpg/policy.py

```diff
@@ -93,16 +93,13 @@ def __init__(
         self.social_vehicle_config = get_social_vehicle_configs(
             **policy_params["social_vehicles"]
         )
-
-        self.social_vehicle_encoder = self.social_vehicle_config["encoder"]
-        self.state_description = get_state_description(
+        self.state_preprocessor = BaselineStatePreprocessor(
             policy_params["social_vehicles"],
             policy_params["observation_num_lookahead"],
             self.action_size,
         )
-        self.state_preprocessor = StatePreprocessor(
-            preprocess_state, to_2d_action, self.state_description
-        )
+
+        self.social_vehicle_encoder = self.social_vehicle_config["encoder"]
```

Comment:
Yeah, I was wondering about that too. Do you mean just the metadata about the encoder like social_vehicle_config be a part of the StatePreprocessor? Or the actual encoder neural network as well?

If the actual encoder network is part of the state preprocessing, would we have to worry about the encoder network using PyTorch and the network used for the agent using another library like TensorFlow? Or perhaps this would be a problem for the ULTRA user... they would either have to make their own encoder or make their agent use a PyTorch network.

Additionally, if the actual encoder network was used in the state preprocessing to encode the social vehicles before sending it back to the agent, the agent would lose control over this encoder network. By the phrase 'lose control', I mean the agent's policy class could not set the state or properties of the encoder network. For example in the DQN, when it learns, this call to calculate `qs_network_target` without gradient calculations would have no effect on the encoder network. That is, the encoder network would still calculate the gradients since it would be a separate network from the agent's network:
https://github.com/christianjans/SMARTS/blob/505b10939e92f504921e18da3f26f54ca08d1f08/ultra/baselines/dqn/dqn/policy.py#L344-L346
Similarly, these calls to train() and eval() wouldn't set the encoder network to train() or eval() mode, it would just set the agent's network to these modes:
https://github.com/christianjans/SMARTS/blob/505b10939e92f504921e18da3f26f54ca08d1f08/ultra/baselines/dqn/dqn/policy.py#L349
https://github.com/christianjans/SMARTS/blob/505b10939e92f504921e18da3f26f54ca08d1f08/ultra/baselines/dqn/dqn/policy.py#L349

Let me know if I can clarify anything or if something I said isn't correct.

Reply:
the encoders are processed to work with pytorch, and the encoding is not happening in the statepreprocessing, statepreprocessor only extracts the vehicle information to feed the networks... lets not touch it for now, make another task to brainstorm the encoders later

Reply:
Will do 👍